aiida_kkr/workflows/imp_BdG.py

```diff
@@ -0,0 +1,414 @@
+            if 'kkr' in self.inputs.imp_scf.gf_writeout:
+                builder.kkr = self.inputs.imp_scf.gf_writeout.kkr
+            if 'params_kkr_overwrite' in self.inputs.imp_scf.gf_writeout:
+                builder.params_kkr_overwrite = self.inputs.imp_scf.gf_writeout.params_kkr_overwrite
+        builder.gf_writeout.kkr = builder.kkr  # pylint: disable=no-member
+
+        if 'params_overwrite' in self.inputs.imp_scf.scf:
+            builder.scf.params_overwrite = self.inputs.imp_scf.scf.params_overwrite
+
+        imp_calc = self.submit(builder)
+
+        return ToContext(last_imp_calc=imp_calc)
+
+    def imp_BdG_calc(self):
+        """
+        run BdG one-shot impurity calculation
+        """
+
+        builder = kkr_imp_wc.get_builder()
+        if 'impurity_info' in self.inputs:
+            builder.impurity_info = self.inputs.impurity_info
+        builder.voronoi = self.inputs.voronoi
+        builder.kkr = self.inputs.kkr
+        builder.kkrimp = self.inputs.kkrimp
+
+        if 'startpot' in self.inputs.imp_scf:
+            builder.startpot = self.inputs.imp_scf.startpot
+        else:
+            builder.startpot = self.ctx.last_imp_calc.outputs.converged_potential
+
+        if 'remote_data_gf' in self.inputs.BdG_scf:
+            builder.remote_data_gf = self.inputs.BdG_scf.remote_data_gf
+
+        if 'gf_writeout' in self.inputs.BdG_scf:
+            if 'kkr' in self.inputs.BdG_scf.gf_writeout:
+                builder.kkr = self.inputs.BdG_scf.gf_writeout.kkr
+            if 'params_kkr_overwrite' in self.inputs.BdG_scf.gf_writeout:
+                builder.params_kkr_overwrite = self.inputs.BdG_scf.gf_writeout.params_kkr_overwrite
+        if 'kkr' in self.inputs:
+            builder.gf_writeout.kkr = builder.kkr  # pylint: disable=no-member
+
+        builder.remote_data_host = self.inputs.BdG_scf.remote_data_host
+
+        if 'options' in self.inputs.BdG_scf:
+            builder.options = self.inputs.BdG_scf.options
+        else:
+            if 'options' in self.inputs.imp_scf:
+                builder.options = self.inputs.imp_scf
+            else:
+                builder.options = self.inputs.options
+
+        settings = kkr_imp_wc.get_wf_defaults()[1]
+        settings['strmix'] = 0.01
+        settings['aggrmix'] = 0.01
+        settings['nsteps'] = 1
+        settings['kkr_runmax'] = 1
+        settings['mag_init'] = True
+        builder.wf_parameters = Dict(dict=settings)
+        BdG_params = self.inputs.BdG_settings.get_dict()
+        builder.scf.params_overwrite = Dict(dict=BdG_params)  # pylint: disable=no-member
+
+        imp_calc_BdG = self.submit(builder)
+
+        return ToContext(last_imp_calc_BdG=imp_calc_BdG)
+
+    def do_BdG_scf(self):
+        """
+        run BdG scf step only if BdG impurity potential not provided and DOS calculation is not planned
+        """
+        if (not 'startpot' in self.inputs.BdG_scf) and (not self.inputs.calc_DOS):
+            return True
```

Comment:
is this a placeholder for the impurity BdG scf step (which is not implemented yet)?